index.js

```diff
@@ -36,11 +37,40 @@ var reservedRe = /^\.+$/;
 var windowsReservedRe = /^(con|prn|aux|nul|com[0-9]|lpt[0-9])(\..*)?$/i;
 var windowsTrailingRe = /[\. ]+$/;
 
-function sanitize(input, replacement) {
+function containsInvalids(input, invalids) {
+  for (var i = 0; i < invalids.length; i++) {
+    if (input.indexOf(invalids[i]) !== -1) {
+      return true;
+    }
+  }
+  return false;
+}
+
+function sanitize(input, replacement, invalids) {
   if (typeof input !== 'string') {
     throw new Error('Input must be string');
   }
-  var sanitized = input
+  if (typeof replacement === 'string' && invalids.indexOf(replacement) !== -1) {
+    throw new Error("The replacement string can't be part of options.additionalInvalidStrings or contain substrings which are part of options.additionalInvalidStrings!");
+  }
+  var sanitized = input;
+  var counter = 0;
+  while (containsInvalids(sanitized, invalids)) {
+    for (var i = 0; i < invalids.length; i++) {
+      if (typeof replacement !== 'string') {
+        var tempReplacement = replacement(invalids[i]);
+        sanitized = sanitized.split(invalids[i]).join(tempReplacement);
```

Comment:
The replace function uses a RegExp to find the parts to be replaced. Passing for example `|` as an invalid character will cause an error, because the pipe character is a reserved character in RegEx.

I used a RegExp for this at first as well, but had to change it for this reason. Not sure if it will work when simply passing a string though?

Reply:
You can pass the strings directly to `replace`:

```js
console.log('a|a'.replace('|', ''));
// => 'aa'
```

Reply:
Nice! I'll change the code :D

Reply:
From MDN:  
> If `pattern` is a string, only the first occurrence will be replaced.

I guess now we'd have to add another while loop and our performance improvement will be gone. Damn...

Reply:
Haha, I'm not even sure I fully understood that failure but if all else fails we can document our way out of it. 

But frankly, with all the issues coming up, I'm having a hard time backing the string-array implementation. Not that I'm shying away from hard work, but rather because I think the right way to implement this feature is by allowing users to pass in a regex. I've thought about it some more and I don't think we'd be introducing an extra attack vector. Calling it out in the documentation is also good enough in my opinion.

Reply:
How about we add both?  
We could make the users aware of the RegEx attack problem and call the string-array part and 'experimental' feature? Then see if this causes many issues or not?  
I know what you're getting at with the RegEx, but I just really want to have the simplicity of strings...

Reply:
I understand the motivation to keeping it simple, for me it just doesn't equate to the cost of maintaining it -- having both implementations would be an even greater cost. 

I don't want to be a blocker for this, @parshap what do you think?